src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java

```diff
@@ -2808,44 +2808,152 @@ public void setHearingSelectedInfoForTask(List<Element<HearingData>> ordersHeari
+            if (selectedDraftOrder != null && (standardDirectionsOrder.equals(selectedDraftOrder.getOrderType()))) {
+                log.info("sdo EaA");
+                isSdoOrder = true;
+                sdoHearings = buildSdoHearingsListFromSdoDetails(selectedDraftOrder.getSdoDetails());
+            }
+
             if (ManageOrdersUtils.isOrderEdited(caseData, eventId)) {
-                setHearingSelectedInfoForTask(caseData.getManageOrders().getOrdersHearingDetails(), caseDataUpdated);
+                log.info("during edit -->");
+                setHearingSelectedInfoForTask(isSdoOrder ? sdoHearings : caseData.getManageOrders().getOrdersHearingDetails(), caseDataUpdated);
                 String isOrderApproved = isOrderApproved(caseData, caseDataUpdated, performingUser);
-                setIsHearingTaskNeeded(caseData.getManageOrders().getOrdersHearingDetails(),
+                setIsHearingTaskNeeded(isSdoOrder ? sdoHearings : caseData.getManageOrders().getOrdersHearingDetails(),
                                        caseDataUpdated,isOrderApproved,amendOrderCheckEnum,eventId);
+                log.info("caseDataUpdated while EaA--Edit--> {}",caseDataUpdated);
             } else {
                 UUID selectedOrderId = elementUtils.getDynamicListSelectedValue(
                     caseData.getDraftOrdersDynamicList(), objectMapper);
 
                 if (null != caseData.getDraftOrderCollection()) {
                     for (Element<DraftOrder> e : caseData.getDraftOrderCollection()) {
+                        log.info("during non edit");
                         DraftOrder draftOrder = e.getValue();
                         if (e.getId().equals(selectedOrderId)) {
-                            setHearingSelectedInfoForTask(draftOrder.getManageOrderHearingDetails(), caseDataUpdated);
+                            log.info("during non edit.....");
+                            setHearingSelectedInfoForTask(isSdoOrder ? sdoHearings : draftOrder.getManageOrderHearingDetails(), caseDataUpdated);
                             String isOrderApproved = isOrderApproved(caseData, caseDataUpdated, performingUser);
-                            setIsHearingTaskNeeded(draftOrder.getManageOrderHearingDetails(),
+                            setIsHearingTaskNeeded(isSdoOrder ? sdoHearings : draftOrder.getManageOrderHearingDetails(),
                                                    caseDataUpdated,isOrderApproved,amendOrderCheckEnum,eventId);
                         }
                     }
+                    log.info("caseDataUpdated while EaA--non-Edit--> {}",caseDataUpdated);
                 }
             }
         }
     }
 
+
+    private List<Element<HearingData>> buildSdoHearingsListFromStandardDirectionOrder(StandardDirectionOrder sdo) {
+
+        List<Element<HearingData>> sdoHearingsList = new ArrayList<>();
+        if (null != sdo) {
+            if (null != sdo.getSdoSecondHearingDetails()) {
```

Comment:
Should you also check for radio button selection rather than just checking !=null